services/redis/queue.js

```diff
@@ -59,21 +59,30 @@ const registerV2ServiceQueue = async (token, userId, follows, topics, locations,
   }
 };
 
+const setRequiredTime = (delay) => {
+  let requiredTime = momentTz().tz('America/Los_Angeles');
+  if (delay === 0) {
+    const randomTime = Math.floor(Math.random() * (40 - 10 + 1)) + 10;
+    requiredTime = momentTz().tz('America/Los_Angeles').add(randomTime, 'minutes');
+  } else {
+    const randomTime = sample([6, 7, 8]);
+    const additionalDays = delay;
+    requiredTime = momentTz()
+      .tz('America/Los_Angeles')
+      .set({hour: randomTime})
+      .add(additionalDays, 'days');
+  }
+  return requiredTime;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setRequiredTime` function seems to be working as expected. However, it's not clear why the time zone is hardcoded to 'America/Los_Angeles'. This could lead to unexpected behavior if the system is used in different time zones. Consider passing the time zone as a parameter to the function or retrieving it from a configuration file or environment variable.
```diff
-  let requiredTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz().tz(timeZone);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->